packages/neo4j/src/index.ts

```diff
@@ -0,0 +1,258 @@
+
+    async createUser({ emailVerified, ...userData }) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` CREATE (u:User)
+          SET
+            u += $userData
+            ${
+              undefined !== emailVerified
+                ? `, u.emailVerified = datetime($emailVerified)`
+                : ``
+            }
+          RETURN properties(u) AS u`,
+        {
+          userData: { id: uuid(), ...userData },
+          emailVerified,
+        }
+      )
+    },
+
+    async getUser(id) {
+      return await neo4jWrap(
+        neo4jSession,
+        `MATCH (u:User { id: $id }) RETURN properties(u) AS u`,
+        {
+          id,
+        },
+        {
+          tx: "read",
+        }
+      )
+    },
+
+    async getUserByEmail(email) {
+      return await neo4jWrap(
+        neo4jSession,
+        `MATCH (u:User { email: $email }) RETURN properties(u) AS u`,
+        {
+          email,
+        },
+        {
+          tx: "read",
+        }
+      )
+    },
+
+    async getUserByAccount(provider_providerAccountId) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` MATCH (u:User)-[:HAS_ACCOUNT]->(a:Account {
+            provider: $provider,
+            providerAccountId: $providerAccountId
+          })
+          RETURN properties(u) AS u`,
+        {
+          ...provider_providerAccountId,
+        },
+        {
+          tx: "read",
+        }
+      )
+    },
+
+    async updateUser({ id, emailVerified, ...userData }) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` MATCH (u:User { id: $id })
+          SET
+            u += $userData
+            ${
+              undefined !== emailVerified
+                ? `, u.emailVerified = datetime($emailVerified)`
+                : ``
+            }
+          RETURN properties(u) AS u`,
+        {
+          id,
+          userData,
+          emailVerified,
+        }
+      )
+    },
+
+    async deleteUser(id) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` MATCH (u:User { id: $id })
+          WITH u, properties(u) AS properties
+          DETACH DELETE u
+          RETURN properties AS deletedUser`,
+        {
+          id,
+        }
+      )
+    },
+
+    // * * * * * * * * * * * * * * * *
+    // Account
+    // * * * * * * * * * * * * * * * *
+
+    async linkAccount({ userId, providerAccountId, provider, ...accountData }) {
+      return await neo4jWrap(
+        neo4jSession,
+        ` MATCH (u:User { id: $userId })
+          MERGE (a:Account { 
+            providerAccountId: $providerAccountId, 
+            provider: $provider 
+          }) 
+          ON CREATE SET 
+            a.id = $id
+          SET 
+            a += $accountData
+          MERGE (u)-[:HAS_ACCOUNT]->(a)
+          RETURN a { .*, userId: u.id } AS a`,
+        {
+          id: uuid(),
+          userId,
+          providerAccountId,
+          provider,
+          accountData,
+        }
```

Comment:
Technically, everything here is `accountData`, could you explain the reason for this property, rather than using the spread operator?

Reply:
I see what you mean but there is a reason.
* On this line I am setting account data to be everything except for userId, providerAccountId, provider.
  `async linkAccount({ userId, providerAccountId, provider, ...accountData }) {`

* Finding or creating the Account node.
  ```
  MERGE (a:Account { 
    providerAccountId: $providerAccountId, 
    provider: $provider 
  }) 
  ```

* Setting the account data here
  `SET a += $accountData`
  userId should not be set on this node because a relationship to a user node is created
  and, it is unnecessary to set providerAccountId and provider properties again.